src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -238,6 +239,182 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
+        {
+            internal const string Version = "1.0.0.0";
+            internal const string InformationalVersion = "1.0.0.0";
+        }
+
+        [ResourceConsumption(ResourceScope.Machine, ResourceScope.Machine)]
+        [ResourceExposure(ResourceScope.None)]
+        static string GenerateAssemblyId(Type type)
+        {
+            Module[] modules = type.Assembly.GetModules();
+            ArrayList list = new ArrayList();
+            for (int i = 0; i < modules.Length; i++)
+            {
+                list.Add(modules[i].ModuleVersionId.ToString());
+            }
+            list.Sort();
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < list.Count; i++)
+            {
+                sb.Append(list[i].ToString());
+                sb.Append(",");
+            }
+            return sb.ToString();
+        }
+        internal static FileStream GenerateSerializerFile(XmlMapping[] xmlMappings, Type[] types, string defaultNamespace, Assembly assembly, Hashtable assemblies, string codePath)
+        {
+            Compiler compiler = new Compiler();
+            try
+            {
+                Hashtable scopeTable = new Hashtable();
+                foreach (XmlMapping mapping in xmlMappings)
+                    scopeTable[mapping.Scope] = mapping;
+                TypeScope[] scopes = new TypeScope[scopeTable.Keys.Count];
+                scopeTable.Keys.CopyTo(scopes, 0);
+
+                assemblies.Clear();
+                Hashtable importedTypes = new Hashtable();
+                foreach (TypeScope scope in scopes)
+                {
+                    foreach (Type t in scope.Types)
+                    {
+                        compiler.AddImport(t, importedTypes);
+                        Assembly a = t.Assembly;
+                        string name = a.FullName;
+                        if (assemblies[name] != null)
+                            continue;
+                        if (!a.GlobalAssemblyCache)
+                        {
+                            assemblies[name] = a;
+                        }
+                    }
+                }
+                for (int i = 0; i < types.Length; i++)
+                {
+                    compiler.AddImport(types[i], importedTypes);
+                }
+                compiler.AddImport(typeof(object).Assembly);
+                compiler.AddImport(typeof(XmlSerializer).Assembly);
+
+                IndentedWriter writer = new IndentedWriter(compiler.Source, false);
+
+                writer.WriteLine("#if _DYNAMIC_XMLSERIALIZER_COMPILATION");
+                writer.WriteLine("[assembly:System.Security.AllowPartiallyTrustedCallers()]");
+                writer.WriteLine("[assembly:System.Security.SecurityTransparent()]");
+                writer.WriteLine("[assembly:System.Security.SecurityRules(System.Security.SecurityRuleSet.Level1)]");
+                writer.WriteLine("#endif");
```

Comment:
Line 305 ~ line 309 are not necessary on NetCore as security attributes are no-ops on NetCore. We can remove them.